utils/manifest/_internal/parser.py

```diff
@@ -155,14 +166,16 @@ def weblog_declaration(n: str, e: dict[str, Any], component: str) -> Return:
         new_entries: list[Condition] = []
         all_weblogs: list[str] = []
         for weblog in e[n]:
+            if not isinstance(weblog, str):
```

Comment:
Is there any valid reason for `weblog` not being a `str` ? If no, could you raise a `TypeError`?